### ISSUE TITLE
IndexError when processing surface NaNs

### ISSUE THREAD

**Issue by MayQueen** [dongqi-DQ/WRF4PALM#21]
```
Geostrophic wind estimation...
100%|████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00,  8.61it/s]
Resolving surface NaNs...
  0%|                                                                                            | 0/5 [00:03<?, ?it/s]
multiprocess.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "C:\Users\ZCWang\AppData\Roaming\Python\Python39\site-packages\multiprocess\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\ZCWang\Downloads\WRF4PALM_v1.1\dynamic_util\surface_nan_solver.py", line 55, in solve_surface
    ds_we[var].loc[dict(time=ts)][:,j,bc] = surface_nan_uv(ds_we[var].sel(time=ts)[:,j,bc].data,z,surface_value)
  File "C:\Users\ZCWang\Downloads\WRF4PALM_v1.1\dynamic_util\surface_nan_solver.py", line 19, in surface_nan_uv
    a = (data[nan_idx]-uv10)/np.log(z[nan_idx]-10)
IndexError: index 200 is out of bounds for axis 0 with size 200
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\ZCWang\Downloads\WRF4PALM_v1.1\run_config_wrf4palm.py", line 704, in <module>
    main()
  File "C:\Users\ZCWang\Downloads\WRF4PALM_v1.1\run_config_wrf4palm.py", line 507, in main
    pool_outputs = list(
  File "C:\Users\ZCWang\AppData\Roaming\Python\Python39\site-packages\tqdm\std.py", line 1195, in __iter__
    for obj in iterable:
  File "C:\Users\ZCWang\AppData\Roaming\Python\Python39\site-packages\multiprocess\pool.py", line 870, in next
    raise value
IndexError: index 200 is out of bounds for axis 0 with size 200
```
Error happend with `python=3.9.12` ,`numpy=1.22.4`. Looking forward to your help, Thanks.
My namelist input is attached to the issue.
[namelist.wrf4palm.txt](https://github.com/dongqi-DQ/WRF4PALM/files/10022366/namelist.wrf4palm.txt)

**Comment by dongqi-DQ** [comment]
Can you provide a sample of your wrf output? So I can try to replicate this issue. This could be the domain locating issue we had before (#11).

This error sometimes is due to high terrain in WRF located at PALM's domain lateral boundaries. But your namelist looks fine as the domain height of PALM is 2000 m.

**Comment by dongqi-DQ** [comment]
And I also noticed that you're using v1.1. Can you try to update to the latest version of WRF4PALM and see if the issue is solved?

**Comment by MayQueen** [comment]
Thank you for your feedback. After I update `WRF4PALM-1.1` to `WRF4PALM-1.1.1_beta`, the `surface NaNs indexerror` was solved. However, I get the `nan` values for `soil_moisture` and `deep_soil_temperature`. Currently I do not know whether it will affect the calculation of PALM.
```
Resolving surface NaNs...
100%|████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:11<00:00,  2.35s/it]
Writing NetCDF file
Add to your *_p3d file:
 soil_temperature = [273.1600036621094, 273.1600036621094, 273.1600036621094, 273.1600036621094, 273.1600036621094, 273.1600036621094, 273.1600036621094, 273.1600036621094]
 soil_moisture = [nan, nan, nan, nan, nan, nan, nan, nan]
 deep_soil_temperature = nan

PALM dynamic input file is ready. Script duration: 0:05:33.831722
Start time: 2017-09-19T12:00:00.000000000
End time: 2017-09-19T14:00:00.000000000
Time step: 3600.0 seconds
```

**Comment by dongqi-DQ** [comment]
Can you check soil moisture in WRF for your PALM domain area? Currently, WRF4PALM fills soil moisture when the entire domain returns 0.0, but does not handle `nan`s.

Again, it would be great if you can provide a sample WRF output file so I can point to the cause of this issue easier.

Thanks

**Comment by MayQueen** [comment]
Thank you again. The update `namelist.wrf4palm` and sample `wrfout` are provided.
namelist input
[namelist.wrf4palm.txt](https://github.com/dongqi-DQ/WRF4PALM/files/10026717/namelist.wrf4palm.txt)

wrfout
https://drive.google.com/drive/folders/1ja7cZvRsznBVdeV46WRMe4NvA_QUoSSl?usp=share_link

**Comment by dongqi-DQ** [comment]
Thanks for your prompt response. I will look into this soon.

**Comment by dongqi-DQ** [comment]
I ran a test and found that your entire PALM domain is over **water surface.** And in WRF4PALM, I used median value of soil moisture and deep soil temperature of **land surface** - to avoid mismatch between WRF and PALM land use classification. But this caused `nan` value when no land is included in PALM domain. I've applied a fix in the latest commit ffbfc90dc5c6729f3f70d30844985192b598b0a2. 

The latest version should work with your WRF output and wrf4palm namelist. 

I will close this issue now, but feel free to reopen it if the issue is not solved. 

Thanks :)
